src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -213,7 +219,14 @@ private SnowflakeTableRow processRecord(SinkRecord record) {
       meta.set(HEADERS, parseHeaders(record.headers()));
     }
 
-    return new SnowflakeTableRow(valueContent, meta);
+    // extract tenantId from the input Kafka record
+    tenantId = extractTenantId(record);
+    // extract entityType from the input Kafka record
+    entityType = extractEntityType(record);
+    // extract rowCreated from the input Kafka record
+    rowCreated = extractRowCreated(record);
+
```

Comment:
Just a suggestion - instead of 3 different method call on same record, can we extract all three items in one go and use it?

Reply:
Incorporated, thanks!